app/models/post.rb

```diff
@@ -0,0 +1,22 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments
+  has_many :likes
+
+  # attribute
+  attribute :title, :string
+  attribute :text, :text
+  attribute :comments_count, :integer, default: 0
+  attribute :likes_count, :integer, default: 0
+
+  # callback
+  after_save :update_user_posts_counter
+
+  def update_user_posts_counter
+    author.update(posts_count: author.posts.count)
+  end
+
```

Comment:
- [ ] **[OPTIONAL]** I would recommend you make this method private in keeping with encapsulation and to avoid other parts of the app from calling this method since it is local to this class.